### DISCUSSION TITLE
How to test speed through socks5 proxy?

### DISCUSSION THREAD

**Discussion by paolosezart** [esnet/iperf#1660] [General]
I have studied the commands of this wonderful utility, but did not find any way to measure internet speed via socks5 proxy.
Please, write with which command I can do it.
Thanks!

**Comment by davidBar-On** [comment]
@paolosezart, I am not familiar with socks5, but see the discussion in issue #1095.  It might help.

**Comment by paolosezart** [comment]
@davidBar-On, thanks. Although it is very strange that a network tool developed by professionals does not have such an obvious thing as support for working through a proxy.

**Comment by davidBar-On** [comment]
@paolosezart, I have submitted PR #1676 with TCP support for SOCKS5 Proxy (based on @mister-goo [code](https://github.com/esnet/iperf/issues/1095#issuecomment-1146514695)).  Are you able to build this version and see if it works for you?  This will be appreciated.

**Reply by sergeevabc** [reply]
Alas, I have no skills to build it myself, bit I am willing to test it on Windows 7 x64.
